### PR TITLE
Merge c_emailVerificationV2 into c_emailVerification

### PR DESCRIPTION
**Pull request by abuzar-deriv**
1. Main change:
    - Added optional param `isViaAPI` in `c_emailVerification` to run task for api account creation
    - Updated the locator to get verification link to use arrow function to reduce duplicate code
    - Handled scenario where credentials were not rotated if c_emailVerification throws error
2. Minor: added `isMT5ResetPassword: true,` in e2e/smoke/Wallets/mt5ResetPasswordResponsive.cy.js

### CODE REVIEW

File: cypress/support/commands/common.js [hunk 1]
```diff
@@ -337,61 +338,53 @@ Cypress.Commands.add(
...
-                    const code = href.match(/code=([A-Za-z0-9]{8})/)
-                    verification_code = code[1]
-                    Cypress.env('walletsWithdrawalCode', verification_code)
-                    cy.log('Verification link found')
-                  } else {
-                    cy.log('Verification link not found')
-                  }
-                })
-            }
+            ;(() =>
```
Comment (bilalmajid-deriv) [inline cypress/support/commands/common.js:353]: is the ; required

**Review by bilalmajid-deriv — COMMENTED** [review]
one small comment

File: cypress/support/commands/common.js (hunk 1 above)
Comment (abuzar-deriv) [reply cypress/support/commands/common.js:353]: Yes. Without ; it was executing the function with cy.wrap(verificationEmail).click().

Comment (bilalmajid-deriv) [reply cypress/support/commands/common.js:353]: ah okay thought it was a typo

**Review by bilalmajid-deriv — APPROVED** [review]

**Review by markodonnell-deriv — APPROVED** [review]
Looks good

**Pull request merged**